### PR TITLE
Add vocabulary terms to Require Explicit Binding

### LINKED ISSUE #2611: Add support for the proposed explicit binding annotations
The OASIS TC has proposed [two new annotations](https://issues.oasis-open.org/browse/ODATA-1538?jql=project%20%3D%20ODATA) for binding functions and actions to particular paths instead of particular types. We should add those annotations to ODL, and we should update our logic to use these annotations to validate URLs are restricted to these path bindings. 

We should create a feature request against AGS when this work is done so that they can take the change there and support it on graph.

### PR DESCRIPTION
**Pull request by lisicase**
### Issues

This pull request fixes #2611.

### Description

Adds two new vocabulary annotation terms: RequiresExplicitBinding and ExplicitOperationBindings.

### Checklist

- [x] Test cases added
- [x] Build and test with one-click build and test script passed

### CODE REVIEW

File: src/Microsoft.OData.Edm/Vocabularies/CoreVocabularies.xml [hunk 1]
```diff
@@ -443,6 +443,14 @@ Any simple identifier | Any type listed in `Validation.OpenPropertyTypeConstrain
     <Annotation Term="Core.LongDescription" String="The annotation value will usually be an expression, e.g. using properties of the binding parameter type for instance-dependent availability, or using properties of a singleton for global availability. The static value `null` means that availability cannot be determined upfront and is instead expressed as an operation advertisement." />
   </Term>
 
+  <Term Name="RequiresExplicitBinding" Type="Core.Tag" DefaultValue="true" AppliesTo="Action Function">
+	<Annotation Term="Core.Description" String="This bound action or function is only available on model elements annotated with the ExplicitOperationBindings term." />
+  </Term>
+
+  <Term Name="ExplicitOperationBindings" Type="Collection(Core.QualifiedBoundOperationName)">
```
Comment (mikepizzo) [inline src/Microsoft.OData.Edm/Vocabularies/CoreVocabularies.xml:450]: >Core.QualifiedBoundOperationName [](http://example.com/codeflow?start=58&length=32)

The definition of this term also needs to be added to the PR

File: src/Microsoft.OData.Edm/Vocabularies/CoreVocabularies.xml [hunk 2]
```diff
@@ -443,6 +443,14 @@ Any simple identifier | Any type listed in `Validation.OpenPropertyTypeConstrain
     <Annotation Term="Core.LongDescription" String="The annotation value will usually be an expression, e.g. using properties of the binding parameter type for instance-dependent availability, or using properties of a singleton for global availability. The static value `null` means that availability cannot be determined upfront and is instead expressed as an operation advertisement." />
   </Term>
 
+  <Term Name="RequiresExplicitBinding" Type="Core.Tag" DefaultValue="true" AppliesTo="Action Function">
+	<Annotation Term="Core.Description" String="This bound action or function is only available on model elements annotated with the ExplicitOperationBindings term." />
+  </Term>
+
+  <Term Name="ExplicitOperationBindings" Type="Collection(Core.QualifiedBoundOperationName)">
+    <Annotation Term="Core.Description" String="The qualified names of explicitly bound operations that are supported on the target model element. These operations are in addition to any operations not annotated with RequiresExplicitBinding that are bound to the type of the target model element." />
+  </Term>
```
Comment (mikepizzo) [inline src/Microsoft.OData.Edm/Vocabularies/CoreVocabularies.xml:452]: Note that, in the final version of the OASIS PR, the Description was updated and a LongDescription was added.

File: test/FunctionalTests/Microsoft.OData.Edm.Tests/Vocabularies/CoreVocabularyTests.cs [hunk 3]
```diff
@@ -378,6 +384,19 @@ public void TestBaseCoreVocabularyModel()
...
+            Assert.NotNull(requiresExplicitBindingTerm);
+            var requiresExplicitBindingType = requiresExplicitBindingTerm.Type.Definition as IEdmTypeDefinition;
+            Assert.NotNull(requiresExplicitBindingType);
+            Assert.Equal(EdmPrimitiveTypeKind.Boolean, requiresExplicitBindingType.UnderlyingType.PrimitiveKind);
+
+            var explicitOperationBindingsTerm = coreVocModel.FindTerm("Org.OData.Core.V1.ExplicitOperationBindings");
+            Assert.NotNull(explicitOperationBindingsTerm);
+            var explicitOperationBindingsType = explicitOperationBindingsTerm.Type.Definition;
+            Assert.NotNull(explicitOperationBindingsType);
+            Assert.Equal("Collection(Org.OData.Core.V1.QualifiedBoundOperationName)", explicitOperationBindingsType.FullTypeName());
```
Comment (mikepizzo) [inline test/FunctionalTests/Microsoft.OData.Edm.Tests/Vocabularies/CoreVocabularyTests.cs:397]: >QualifiedBoundOperationName [](http://example.com/codeflow?start=55&length=27)

Once you add the QualifiedBoundOperationName, you'll want to add a test to retrieve and validate that term and then verify that explicitOperationBindingsType is of that type.

**Review by mikepizzo — CHANGES_REQUESTED** [review]
:clock1:

**Review by mikepizzo — APPROVED** [review]
:shipit:

**Review by xuzhg — APPROVED** [review]

**Pull request merged**